### DISCUSSION TITLE
Error on mac gpu with ot.gromov.fgw_barycenters

### DISCUSSION THREAD

**Discussion by youssef62** [PythonOT/POT#570] [Q&A] [answered]
Hello, 
Is it really necessary to have .double in this line ? 
(`backend.py` line 1944) 
 ```
   def norm(self, a, axis=None, keepdims=False):
        return torch.linalg.norm(a.double(), dim=axis, keepdims=keepdims)

```
It has the effect of coverting to float64 which is not supported by mps (gpu for mac silicon)  , which raises this error 
`TypeError: Cannot convert a MPS Tensor to float64 dtype as the MPS framework doesn't support float64. Please use float32 instead.`

Is this conversion to float64 intended ?

**Comment by rflamary** [comment]
I donnt think it is necessary and i don't remember why we have this. This is a recent change

@kachayev you did this change in the backend, do you knwo why? We should try to keep the precision and not change it there.

**Reply by rflamary** [reply, marked as answer]
We rmoved this conversion in this PR that is now merged.
https://github.com/PythonOT/POT/pull/572

**Reply by youssef62** [reply]
Thank you for the quick fix.
